Derek task4 sprint2: Dynamic Automation of IP Address

elb-dev opened the pull request:
# What:
Creates a local IP address for use in the cloud
# How to use:
## Local testing:

Prerequisite: Powershell

- Open a Powershell terminal
- Navigate to automation/ip-address-script
- Run ip-create.ps1
- Optional: Include desired parameters in the run command, example: `./create-ip.ps1 -Tenant #### -Subscription #### -ResourceGroupName resource-group-name -IpName myIpAddress`
- Follow instructions with subscription id, tenant id, username, and password
- Most parameters not required. Those not included in run command which are will be requested. All else will be set from default values.

## Why:
Cloud projects need an IP address, and it needs to be easily deployable

# Time:
1 hour researching
4 hours writing and testing
2 hours testing and reviewing team code

OyunominJuno committed:
Added the version 1 script

OyunominJuno committed:
added template files

OyunominJuno committed:
added another layer of directory

OyunominJuno committed:
Merge pull request #1 from OyunominJuno/derek-task5

Creates ip creation automation

OyunominJuno committed:
Merge pull request #2 from OyunominJuno/jg-task4

Jg task4 virtual machine automation script

OyunominJuno committed:
Merge pull request #3 from OyunominJuno/jenv-task2

Jenv task2

OyunominJuno committed:
Merge pull request #4 from North-Seattle-College/development

updating the repo from the NSC repo

OyunominJuno committed:
wrong commit - deleting the file

OyunominJuno committed:
Merge pull request #5 from OyunominJuno/OyunominJuno-patch-1

wrong commit - deleting the file

elb-dev committed:
Adds the parameter and template files

Parameters should be removed later.

elb-dev committed:
Update create-ip.ps1

elb-dev committed:
Sets correct login

elb-dev committed:
Makes the ip address far more dynamic

elb-dev committed:
Updates documentation

asiagromasia commented:
Adding allowed options to IpVersion, server location and method would make the code more safe.

asiagromasia commented on automation/ip-address-script/template.json:
```diff
@@ -0,0 +1,40 @@
+            "type": "String"
+        },
+        "serverLocation": {
+            "defaultValue": "westus2",
```
You could add here AllowedValues to limit the input possibilities

sunshinejen replied:
Good idea!

asiagromasia commented on automation/ip-address-script/template.json:
```diff
@@ -0,0 +1,40 @@
+            "type": "string"
+        },
+        "IpVersion": {
+            "defaultValue": "IPv4",
```
add allowValues here would be nice as well

asiagromasia commented on automation/ip-address-script/template.json:
```diff
@@ -0,0 +1,40 @@
+            "type": "string"
+        },
+        "IpMethod": {
+            "defaultValue": "Dynamic",
```
add option here would limit the input possibilities

asiagromasia commented:
adding output parameter to template.json would help with referencing this resource

sunshinejen reviewed (approved):
tested and verified deployment to Azure

elb-dev committed:
Adds allowed values

Also makes it so it actually creates the ip address in the new resource group if there is one.
